### ISSUE TITLE
Key hashing uses overlapping offsets

### ISSUE THREAD

**Issue by osuushi** [blockai/advisory-lock#5]
Hey there, noticed a small error that means your lock keys are effectively only 5 bytes, instead of the potential 8. The problem is in `strToKey`, where you do:

```js
return [buf.readInt32LE(0), buf.readInt32LE(1)]
```

The parameter there is the _byte_ offset, not the sizeof(int32) offset. So what you actually want is:

```js
return [buf.readInt32LE(0), buf.readInt32LE(4)]
```

To see what I mean, try this:

```js
let buf = Buffer.from([0xff, 0xff, 0, 0, 0, 0, 0, 0])
console.log(buf.readInt32LE(0)) // --> 65535
console.log(buf.readInt32LE(1)) // --> 255
```

So you're getting bytes 0-3 in your first int, and 1-4 in your second int. What you really want is 0-3 in the first, and 4-7 in the second. In most cases, this probably doesn't make much difference, but it does increase the chances of a collision by quite a lot in theory.

Cheers!

**Comment by olalonde** [comment]
Oh ouch, good catch!

**Comment by olalonde** [comment]
Fixed in `advisory-lock@1.1.1`, thanks for the detailed report.
